src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -447,6 +457,42 @@ func (a *StorageMinerActorCode_I) SubmitPoSt(rt Runtime, postSubmission poster.P
 	UpdateRelease(rt, h, st)
 
 	return rt.SuccessReturn()
+
+}
+
+// Called by StoragePowerConsensus subsystem after verifying the Election proof
+// and verifying the PoSt proof within the block. (this happens outside the VM)
+// Assume ElectionPoSt has already been successfully verified when the function gets called.
+func (a *StorageMinerActorCode_I) SubmitElectionPoSt(rt Runtime, postSubmission poster.PoStSubmission) InvocOutput {
+
```

Comment:
add a `//TODO() // TODO: validate caller` in here too. 
also state that:
- the caller MUST be the miner who won the block (who won the block should be callable as a a VM runtime call)
---
- we also need to enforce that this call happens only once per block, OR make it not callable by special privileged messages

Reply:
cc @jzimmerman see above ^ -- re the need for VM runtime to give knowledge on who the miner on the _block_ was (not just tipset) in the invocation call. (not sure this is the only place, if only place we could deal with it another way)

Reply:
Sounds good. Re: enforcing that it is only called once per block, I think we can just set a field in the state tree to record the epoch of the actor's most recent PoSt submission (analogous to `LastChallengeEpoch_`), and invalidate any `SubmitElectionPoSt` messages that don't advance the counter.

Reply:
We need to define when this implicit message is called. In a tipset with N blocks, are the N `SubmitElectionPoSt` calls the first three calls? Before or after cron? Or does each block's call precede only that block's (tipset-deduplicated) messages? Or all at the end?

Reply:
@anorth this will have to get done when we do the message passing on block verif in VM (scheduled in prio list)